src/Forte.Optimizely.ContentUsage/Api/Features/ContentType/ContentTypeUsagesDB.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using EPiServer.Construction;
+using EPiServer.Core;
+using EPiServer.Data;
+using EPiServer.DataAbstraction;
+using EPiServer.DataAccess.Internal;
+using EPiServer.Framework.Blobs;
+using EPiServer.ServiceLocation;
+using EPiServer.Web;
+
+namespace Forte.Optimizely.ContentUsage.Api.Features.ContentType;
+
+[ServiceConfiguration(Lifecycle = ServiceInstanceScope.Singleton)]
+// ReSharper disable once InconsistentNaming
+public class ContentTypeUsagesDB : ContentDB
+{
+    public ContentTypeUsagesDB(ServiceAccessor<IDatabaseExecutor> databaseHandlerAccessor, IPermanentLinkMapper permanentLinkMapper, ILanguageBranchRepository languageBranchRepository, IPropertyDataFactory propertyDataFactory, IContentTypeRepository contentTypeRepository, IPropertyDefinitionRepository propertyDefinitionRepository, IPropertyDefinitionTypeRepository propertyDefinitionTypeRepository, IContentFactory contentFactory, IBlobFactory blobFactory, PropertyValueConverterFactory propertyValueConverterFactory, DatabaseDateTimeHandler databaseDateTimeHandler) : base(databaseHandlerAccessor, permanentLinkMapper, languageBranchRepository, propertyDataFactory, contentTypeRepository, propertyDefinitionRepository, propertyDefinitionTypeRepository, contentFactory, blobFactory, propertyValueConverterFactory, databaseDateTimeHandler)
+    {
+    }
+
+    public IList<ContentTypeUsageCounter> ListContentTypesUsagesCounters()
```

Comment:
can we return IEnumerable instead?

Reply:
This method eventually returns materialized. When preparing ContentTypeUsagesDB class I based on ContentListDB class (which is used ListContentOfContentType invocation in line 57 in ContentUsageService) and it works similarly.
Maybe I'm wrong but we will get whole query results after line 89, right?
So I don't see a solid reason to making this function a generator.
Or maybe even we should allow a consumer to decide for example how many items they wants (for memory freeing reason)?

Reply:
ah, correct, you're right. I overlooked the fact that you're returning a previously created List. This brings me to the idea of IReadOnlyCollection - what do you think?

Reply:
Ok. We could try it. But I will have to wrap List<ContentTypeUsageCounter> into ReadOnlyCollection to prevent any way for adding new items. (https://youtu.be/7hBPI0xYezo?t=238)
I'm just not sure if it's something very useful.

Reply:
@darekhalecki What do you think about this?

Reply:
@Fifok 
I would return IEnumerable<T> even if there is a List<T> under the hood.

For two reasons:
- currently cannot see any harm of mutating the output in the current implementation. we can  add `AsReadOnly();` in the future when it becomes a concern
- in the future we can change the implementation of `ListContentTypesUsagesCounters` to lazy loaded. with IEnumerable return type this won't be a breaking change.

Reply:
> * in the future we can change the implementation of `ListContentTypesUsagesCounters` to lazy loaded. with IEnumerable return type this won't be a breaking change

But that means we will break YAGNI and hope this will be change in the future.
I think if we want sometime change/refactor this return type to generator then it will be a good moment to return IEnumerable.